src/server/rpc/procedures/roboscape/robot.js

```diff
@@ -379,39 +381,139 @@ Robot.prototype.onMessage = function (message) {
             ' ' + message.toString('hex'));
     }
 };
 
+// handle user commands to the robot (through the 'send' rpc)
+Robot.prototype.onCommand = function(command) {
+    const cases = [
+        {
+            regex: /^is alive$/,
+            handler: () => {
+                this.sendToClient('alive', {});
+                return this.isAlive();
+            }
+        },
+        {
+            regex: /^beep (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.beep(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^set speed (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.setSpeed(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^drive (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.drive(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^get range$/,
+            handler: () => {
+                return this.getRange().then(function (value) {
+                    return value && value.range;
+                });
+            }
+        },
+        {
+            regex: /^get ticks$/,
+            handler: () => {
+                return this.getTicks().then(function (value) {
+                    return value && [value.left, value.right];
+                });
+            }
+        },
+        {
+            regex: /^set key(| -?\d+([ ,]-?\d+)*)$/,
+            handler: () => {
+                var encryption = RegExp.$1.split(/[, ]/);
+                if (encryption[0] === '') {
+                    encryption.splice(0, 1);
+                }
+                return this.setEncryption(encryption.map(Number));
+            }
+        },
+        {
+            regex: /^set total rate (-?\d+)$/,
+            handler: () => {
+                this.setTotalRate(+RegExp.$1);
+            }
+        },
+        {
+            regex: /^set client rate (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.setClientRate(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^set led (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.setLed(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^infra light (-?\d+)[, ](-?\d+)$/,
+            handler: () => {
+                this.infraLight(+RegExp.$1, +RegExp.$2);
+            }
+        },
+        {
+            regex: /^reset seq$/,
+            handler: () => {
+                this.setSeqNum(-1);
+            }
+        },
+        {
+            regex: /^reset rates$/,
+            handler: () => {
+                this.resetRates();
+            }
+        },
+    ];
+
+    let matchingCase = cases.find(aCase => command.match(aCase.regex));
+    if (!matchingCase) return false; // invalid command structure
+
+    let rv = matchingCase.handler();
+    if (rv === undefined) rv = true;
```

Comment:
if the handler doesn't return anything assume that it succeeded and reply with true